include/openenclave/attestation/verifier.h

```diff
@@ -191,24 +191,65 @@ typedef oe_result_t (*oe_verify_claims_callback_t)(
  * found.
- * @param[in] cert_in_der a pointer to buffer holding certificate contents
- *  in DER format
- * @param[in] cert_in_der_len size of certificate buffer above
- * @param[in] claim_verify_callback callback routine for custom claim checking
- * @param[in] arg an optional context pointer argument specified by the caller
- * when setting callback
- * @retval OE_OK on a successful validation
- * @retval OE_VERIFY_FAILED on quote failure
- * @retval OE_INVALID_PARAMETER At least one parameter is invalid
- * @retval OE_FAILURE general failure
- * @retval other appropriate error code
+ *
+ * @deprecated
+ *
+ * @param[in] certificate_in_der A pointer to buffer holding certificate
+ * contents in DER format.
+ * @param[in] certificate_in_der_len Size of certificate buffer above.
+ * @param[in] claim_verify_callback Callback routine for custom claim checking.
+ * @param[in] arg An optional context pointer argument specified by the caller
+ * when setting callback.
+ * @retval OE_OK Successful validation.
+ * @retval OE_VERIFY_FAILED Quote failure.
+ * @retval OE_INVALID_PARAMETER One or more invalid parameters.
+ * @retval OE_FAILURE General failure.
+ * @retval Other appropriate error code.
  */
 oe_result_t oe_verify_attestation_certificate_with_evidence(
-    uint8_t* cert_in_der,
-    size_t cert_in_der_len,
+    uint8_t* certificate_in_der,
+    size_t certificate_in_der_len,
     oe_verify_claims_callback_t claim_verify_callback,
     void* arg);
 
+/**
+ * oe_verify_attestation_certificate_with_evidence_v2
+ *
+ * This function performs a custom validation on the input certificate. This
+ * validation includes extracting an attestation evidence extension from the
+ * certificate before validating this evidence. An optional
+ * claim_verify_callback could be passed in for a calling client to further
+ * validate the claims of the enclave creating the certificate.
+ * OE_FAILURE is returned if the expected certificate extension OID is not
+ * found.
+ *
+ * @experimental
+ *
+ * @param[in] certificate_in_der A pointer to buffer holding certificate
+ * contents in DER format.
+ * @param[in] certificate_in_der_len Size of certificate buffer above.
+ * @param[in] endorsements_buffer A pointer to buffer holding endorsements.
+ * @param[in] endorsements_buffer_size Size of the endorsements buffer.
+ * @param[in] policies A pointer to buffer holding policies.
+ * @param[in] policies_size Size of the policies buffer.
+ * @param[out] claims A double-pointer to buffer holding claims.
+ * @param[out] claims_length Size of the claims buffer if not NULL.
+ * @retval OE_OK Successful validation.
+ * @retval OE_VERIFY_FAILED Quote failure.
+ * @retval OE_INVALID_PARAMETER One or more invalid parameters.
+ * @retval OE_FAILURE General failure.
+ * @retval Other appropriate error code.
+ */
+oe_result_t oe_verify_attestation_certificate_with_evidence_v2(
+    uint8_t* certificate_in_der,
+    size_t certificate_in_der_len,
```

Comment:
```suggestion
    size_t certificate_in_der_size,
```
avoid abbreviations, per OE coding style guidelines